### ISSUE TITLE
Regression from `Handle dashes in no-whitespace-within-word rule` PR #2421

### ISSUE THREAD

**Issue by rrenno** [ember-template-lint/ember-template-lint#2464]
Super awesome effort on #2421 to fix the dash support, however it appears this broke the ability to use digits back-to-back, i.e. 

```javascript
/\d+-\d+/
```

So for example the following in our code base now throws a `no-whitespace-within-word` error whereas before it did not. The `5-7` being the formerly allowed text.
```handlebars
<li>
    <b>Select a subset.</b> For Java/Python/C/C++, we recommend starting with requiring perhaps 5-7 per week.
</li>
```

It's worth notiing that digits with a space pass linting:
```handlebars
<li>
    <b>Select a subset.</b> For Java/Python/C/C++, we recommend starting with requiring perhaps 5 - 7 per week.
</li>
```
Also alphabet characters without a space pass linting: 
```handlebars
<li>
    <b>Select a subset.</b> For Java/Python/C/C++, we recommend starting with requiring perhaps five-seven per week.
</li>
```

**Comment by mydea** [comment]
Hmm, I guess the problem here is that dashes are basically considered "spaces" now by the rule, which means it basically interprets it as `perhaps 5 7 per week`. Whereas before it considered `5-7` one "word". IMHO the "correct" solution is probably to allow/ignore numbers, I'd say?

**Comment by mydea** [comment]
I made a PR https://github.com/ember-template-lint/ember-template-lint/pull/2465 rewriting the implementation to use a regex instead. IMHO this is more robust and also handles this case properly!
